app/topics/shared/topic.service.ts

```diff
@@ -133,6 +133,18 @@ export class TopicService {
   }
 
   /**
+   * Updates a given Status
+   * @param id The Id of the Topic, Status of the topic
+   */
+  public saveStatusofTopic(id: number, status: string) {
+    let data = '';
+    data += 'Status=' + status;
+    return this.cmsApiService.putUrl('/api/Topics/' + id + '/Status', data, {})
+      .toPromise()
+      .then();
+  }
```

Comment:
What happens on an error? you don't catch any exceptions. 
At least a `.catch(this.handleError)` should be there.